sdk/servicebus/Azure.Messaging.ServiceBus/src/Sender/ServiceBusSenderClient.cs

```diff
@@ -0,0 +1,522 @@
+        /// Schedules a message to appear on Service Bus at a later time.
+        /// </summary>
+        /// <param name="message"></param>
+        /// <param name="scheduleEnqueueTimeUtc">The UTC time at which the message should be available for processing</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>The sequence number of the message that was scheduled.</returns>
+        public virtual async Task<long> ScheduleMessageAsync(ServiceBusMessage message, DateTimeOffset scheduleEnqueueTimeUtc, CancellationToken cancellationToken = default)
+        {
+            //this.ThrowIfClosed();
+            message.ScheduledEnqueueTimeUtc = scheduleEnqueueTimeUtc.UtcDateTime;
+            return await Connection.ScheduleMessageAsync(message, RetryPolicy, GetSendLinkName(), cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Cancels a message that was scheduled.
+        /// </summary>
+        /// <param name="sequenceNumber">The <see cref="ServiceBusMessage.SystemPropertiesCollection.SequenceNumber"/> of the message to be cancelled.</param>
+        /// <param name="cancellationToken"></param>
+        public virtual async Task CancelScheduledMessageAsync(long sequenceNumber, CancellationToken cancellationToken = default)
+        {
+            //this.ThrowIfClosed();
+            await Connection.CancelScheduledMessageAsync(sequenceNumber, RetryPolicy, GetSendLinkName(), cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Get the send link name if the send link is already open.
+        /// </summary>
+        /// <returns></returns>
+        private string GetSendLinkName()
+        {
+            string sendLinkName = null;
+            if (InnerSender.SendLink.TryGetOpenedObject(out SendingAmqpLink openedLink))
+            {
+                if (openedLink != null)
+                {
+                    sendLinkName = openedLink.Name;
+                }
+            }
+            return sendLinkName;
+        }
+
+        /// <summary>
+        ///   Sends a set of events to the associated Service Bus entity using a batched approach.  If the size of events exceed the
+        ///   maximum size of a single batch, an exception will be triggered and the send will fail.
+        /// </summary>
+        ///
+        /// <param name="messages">The set of event data to send.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        /// <returns>A task to be resolved on when the operation has completed.</returns>
+        ///
+
+        /// <seealso cref="SendRangeAsync(IEnumerable{ServiceBusMessage}, CancellationToken)" />
+        ///
+        internal virtual async Task SendRangeInternal(
+            IEnumerable<ServiceBusMessage> messages,
+            CancellationToken cancellationToken)
```

Comment:
nit: Indentation is off.

We follow below pattern in EventHubs. I think we should keep the same pattern here as well :)
```csharp
... SendRangeInternal(IEnumerable<ServiceBusMessage>...,
                      CancellationToken...)
```

Reply:
Jesse had a comment about this. Storage and other libraries appear to do it the other way.